rotkehlchen/chain/ethereum/modules/lido/decoder.py

```diff
@@ -0,0 +1,526 @@
+                tx_log.topics[0] == ERC20_OR_ERC721_TRANSFER and
+                tx_log.address == wsteth_evm_address and
+                hex_or_bytes_to_address(tx_log.topics[1]) == ZERO_ADDRESS and  # from
+                hex_or_bytes_to_address(tx_log.topics[2]) == user_address  # to # wsteth
+            ):
+                wsteth_received_amount = asset_normalized_value(
+                    amount=hex_or_bytes_to_int(tx_log.data[:32]),
+                    asset=A_WSTETH.resolve_to_crypto_asset(),
+                )
+
+            if sth_wrapped_amount and wsteth_received_amount:
+                paired_wsteth_ratio = FVal(sth_wrapped_amount) / FVal(wsteth_received_amount)
+                break
+
+        if paired_wsteth_ratio is None:
+            return DEFAULT_DECODING_OUTPUT
+
+        in_event = None
+        out_event = None
+        for event in context.decoded_events:
+            if (
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.asset == A_STETH and
+                event.balance.amount == sth_wrapped_amount and
+                event.address == wsteth_evm_address
+            ):
+                event.event_type = HistoryEventType.DEPOSIT
+                event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                event.counterparty = CPT_LIDO_ETH
+                event.notes = f'Sent {sth_wrapped_amount} {A_STETH.symbol_or_name()} for wrapping at wstETH ratio {paired_wsteth_ratio}'  # noqa: E501
+                out_event = event
+            elif (
+                event.event_type == HistoryEventType.RECEIVE and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.asset == A_WSTETH and
+                event.balance.amount == wsteth_received_amount and
+                event.address == ZERO_ADDRESS
+            ):
+                event.event_type = HistoryEventType.RECEIVE
+                event.event_subtype = HistoryEventSubType.RECEIVE_WRAPPED
+                event.counterparty = CPT_LIDO_ETH
+                event.notes = f'Receive {wsteth_received_amount} {A_WSTETH.symbol_or_name()} in exchange of the deposited {A_STETH.symbol_or_name()} at wstETH ratio {paired_wsteth_ratio}'  # {amount} to be replaced in post decoding  # noqa: E501,
+                in_event = event
+        if in_event and out_event:
+            maybe_reshuffle_events(
+                ordered_events=[out_event, in_event],
+                events_list=context.decoded_events,
+            )
+
+        return DecodingOutput(matched_counterparty=CPT_LIDO_ETH, refresh_balances=True)
+
+    def _decode_lido_transfer_shares_for_wsteth_unwrap(
+            self, context: DecoderContext,
+    ) -> DecodingOutput:
+        """Decode wsteth unwrapping in steth from TransferShares log event"""
+
+        user_address = hex_or_bytes_to_address(context.tx_log.topics[2])
+
```

Comment:
please don't add so many random empty lines

```suggestion
        user_address = hex_or_bytes_to_address(context.tx_log.topics[2])
```